cmds/netcat/netcat.go

```diff
@@ -18,30 +18,57 @@ import (
 
 const usage = "netcat [go-style network address]"
 
-var netType = flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
+var (
+	netType = flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
+	listen  = flag.Bool("l", false, "Listen for connections.")
+	verbose = flag.Bool("v", false, "Verbose output.")
+)
 
 func init() {
 	util.Usage(usage)
 }
 
 func main() {
-	flag.Parse()
 	var c net.Conn
 	var err error
-	if len(flag.Args()) < 1 {
+	if flag.Parse(); len(flag.Args()) != 1 {
 		flag.Usage()
 		os.Exit(1)
 	}
 
-	if c, err = net.Dial(*netType, flag.Args()[0]); err != nil {
-		log.Fatalf("%v\n", err)
+	addr := flag.Args()[0]
+
+	if *listen {
+		ln, err := net.Listen(*netType, addr)
+		if err != nil {
+			log.Fatalln(err)
+		}
+		if *verbose {
+			fmt.Fprintln(os.Stderr, "Listening on", ln.Addr())
```

Comment:
"Listening on " (with space?)

Reply:
No need to:
```
Fprintln formats using the default formats for its operands and writes to w.
Spaces are always added between operands and a newline is appended.
```

Reply:
Huh. Learning things every day... sorry about that :)